### PR TITLE
ORA1377 protect validators from egregious data

### PR DESCRIPTION
**Pull request by br4e**
This is the straightforward solution that encompasses simply checking the size of the input to query and msg server functions that could be unbounded, bringing down a node. I found only two functions that can have unbounded inputs because they take an array of some sort. I checked the input size and throw an error if the input is too large, and chose a size limit that allows for a few thousand items. 

Perhaps controversial is I removed `extraData` from `MsgFundTopic` to prevent arbitrarily large data from being sent there. It is unused, but has a comment saying that it *might* be used in the future.

### CODE REVIEW

File: x/emissions/keeper/msgserver/msg_server_losses.go [hunk 1]
```diff
@@ -12,17 +12,29 @@ import (
...
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/gogo/protobuf/proto"
 )
 
 // Called by reputer to submit their assessment of the quality of workers' work compared to ground truth
 func (ms msgServer) InsertBulkReputerPayload(
 	ctx context.Context,
 	msg *types.MsgInsertBulkReputerPayload,
 ) (*types.MsgInsertBulkReputerPayloadResponse, error) {
+	serializedMsg, err := proto.Marshal(msg)
```
Comment (kpeluso) [inline x/emissions/keeper/msgserver/msg_server_losses.go:23]: same

**Review by kpeluso — CHANGES_REQUESTED** [review]
Kinda wish you had kept `extra_data`. Idea was to maximize backward compatibility later when subtopics better supported. Not big prio to maintain tho.

More importantly you need to need to remember to follow the checklist here when adding new global params:
https://www.notion.so/upshothq/Adding-New-Global-Parameters-Checklist-6216783e3b4f4280a2fda6f3487f0164

File: x/emissions/keeper/msgserver/msg_server_worker_payload.go [hunk 2]
```diff
@@ -221,6 +223,16 @@ func (ms msgServer) VerifyAndInsertForecastsFromTopForecasters(
 // A tx function that accepts a list of forecasts and possibly returns an error
 // Need to call this once per forecaster per topic inference solicitation round because protobuf does not nested repeated fields
 func (ms msgServer) InsertBulkWorkerPayload(ctx context.Context, msg *types.MsgInsertBulkWorkerPayload) (*types.MsgInsertBulkWorkerPayloadResponse, error) {
+	serializedMsg, err := proto.Marshal(msg)
```
[the remark this answers is not in the document]
Comment (br4e) [reply x/emissions/keeper/msgserver/msg_server_worker_payload.go]: yeah I was trying to do that but I couldn't figure out how to get the function argument type to be generic enough. I figured out how to do it though and updated

**Comment by br4e** [comment]
I can keep the extraData if you want. Since that field now introduces a security risk, I thought it would be better to add it later if we need it in an upgrade. Upgrades are costly, but I suspect we will be doing a lot of them at the beginning. 

As for the checklist: 

https://www.notion.so/upshothq/Adding-New-Global-Parameters-Checklist-6216783e3b4f4280a2fda6f3487f0164?pvs=4

`x/emissions/proto/emissions/v1/types.proto` doesn't seem to be used for parameters as far as I can tell, so I didn't update there

**Review by kpeluso — APPROVED** [review]

**Pull request merged**